run-build-functions.sh

```diff
@@ -458,7 +458,7 @@ install_dependencies() {
   if [ -f package.json ]
   then
     restore_cwd_cache node_modules "node modules"
-    if [ -f yarn.lock ]
+    if ([ -f yarn.lock ] && [ ! "$USE_YARN" = "false" ]) || [ "$USE_YARN" = "true" ]
```

Comment:
```suggestion
    if [ "$USE_YARN" = "true" ] || ([ "$USE_YARN" != "false" ] && [ -f yarn.lock ]) 
```

Small performance optimization since checking for file existence is slower than doing a string comparison. So if `USE_YARN` is either `true` or `false`, `&& [ -f yarn.lock ]` will be short-circuited.